test/fixtures/test_grammars/external_unicode_column_alignment/scanner.c

```diff
@@ -0,0 +1,85 @@
+#include <tree_sitter/parser.h>
+#include <wctype.h>
+#include <string.h>
+
+enum {
+  LIST_START,
+  LIST_ITEM,
+  LIST_END
+};
+
+typedef struct {
+  int32_t column;
+} Scanner;
+
+void *tree_sitter_external_unicode_column_alignment_external_scanner_create() {
+  Scanner *scanner = malloc(sizeof(Scanner));
+  *scanner = (Scanner){
+    .column = -1 
+  };
+  return scanner;
+}
+
+void tree_sitter_external_unicode_column_alignment_external_scanner_destroy(void *payload) {
+  free(payload);
+}
+
+unsigned tree_sitter_external_unicode_column_alignment_external_scanner_serialize(
+  void *payload,
+  char *buffer
+) {
+  Scanner *scanner = payload;
+  unsigned copied = sizeof(int32_t);
+  memcpy(buffer, &(scanner->column), copied);
+  return copied;
+}
+
+void tree_sitter_external_unicode_column_alignment_external_scanner_deserialize(
+  void *payload,
+  const char *buffer,
+  unsigned length
+) {
+  Scanner *scanner = payload;
+  scanner->column = -1;
+  if (length > 0) {
+    memcpy(&(scanner->column), buffer, sizeof(int32_t));
+  }
+}
+
+bool tree_sitter_external_unicode_column_alignment_external_scanner_scan(
+  void *payload,
+  TSLexer *lexer,
+  const bool *whitelist
+) {
+  Scanner *scanner = payload;
+  // 9633 is the int equivalent of □ (U+25A1)
+  while (iswspace(lexer->lookahead) || 9633 == lexer->lookahead) {
```

Comment:
Maybe write this integer as `0x25A1`? It's probably still worth having a comment explaining that's the box character, but it would be slightly  more self-explanatory.

Reply:
I actually ran into a lot of difficulty with this, weirdly. Neither `L'□'` nor `'□'` worked, nor `\x25A1`. Even stranger is that it works just fine in my grammar: https://github.com/tlaplus-community/tree-sitter-tlaplus/blob/26bbaf5d07356cacbf21defa9316988bc732c8b3/src/scanner.cc#L517

Reply:
Not a single-quoted character literal; just the plain integer `0x25A1`.

Reply:
Ah sure, that works. I'll need to figure out the various ways of writing a codepoint and when they work/don't work eventually.